src/harvester/wasser_de.rs

```diff
@@ -88,11 +98,47 @@ async fn translate_dataset(dir: &Dir, source: &Source, document: Document) -> Re
         .map(|last_checked| Date::parse(&last_checked, format_description!("[year]-[month]-[day]")))
         .transpose()?;
 
+    let mut contact_name: Vec<String> = Vec::new();
```

Comment:
This code is pretty ugly, and I would like some help how to collect all attributes mentioned here into a Vec if they contain a value

Reply:
I think the easiest way would be to use that a) [`Option` implements `IntoIterator`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#impl-IntoIterator) and b) [`Vec` has an `extend` method taking this](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.extend). (The iterator of an `Option` yields either one or zero elements.)

So it should be possible to write this like

```rust
contact_names.extend(document.contact_name);
contact_names.extend(document.contact_name_rl1);
contact_names.extend(document.contact_name_rl2);
...
```

but I have not tried it out myself yet, so beware for compiler errors...

Reply:
That seems to work, thank you.